vm/src/vm.rs

```diff
@@ -198,13 +198,27 @@ impl VirtualMachine {
             vm.get_none(),
         );
 
-        builtins::make_module(&vm, builtins.clone());
-        sysmodule::make_module(&vm, sysmod, builtins);
+        vm
+    }
+
+    pub fn initialize(&mut self, external: bool) -> &Self {
+        flame_guard!("init VirtualMachine");
+
+        builtins::make_module(self, self.builtins.clone());
+        sysmodule::make_module(self, self.sys_module.clone(), self.builtins.clone());
 
         #[cfg(not(target_arch = "wasm32"))]
-        import::import_builtin(&vm, "signal").expect("Couldn't initialize signal module");
+        import::import_builtin(self, "signal").expect("Couldn't initialize signal module");
 
-        vm
+        if external {
+            // We only include the standard library bytecode in WASI
+            import::init_importlib(self, cfg!(not(target_os = "wasi")))
```

Comment:
I think this `cfg` should be in `src/main.rs`, not here. Then you can just pass `external` to `init_importlib`